### PR TITLE
PostgreSQL

### PR DESCRIPTION
**Pull request by dsteinich**
Before making a pull request
----------------------------

- [ ] Make sure all tests run
- [ ] Update the changelog appropriately

Title
-----------
Brief description of changes. Reference the JIRA ticket if appropriate

Description
-----------
If no JIRA ticket is referenced, describe the changes made. Note anything that you want the reviewers to know while
reviewing your pull request

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

**Comment by dsteinich** [comment]
Some highlights:

- Geom added to several tables so joining to station/station_sum is no longer needed. Oracle spatial indexes where very large and it was impracticable to add them on the larger tables.

- Lower case sorts before upper case.

File: src/main/resources/mybatis/station.xml [hunk 1]
```diff
@@ -32,9 +32,9 @@
                prime.vertical_accuracy_unit,
                prime.elevation_method,
                prime.vdatum_id_code,
-               prime.country_code,
-               prime.state_fips_code,
-               prime.county_fips_code,
+               substring(prime.governmental_unit_code, '[^:]+') country_code,
+               substring(prime.governmental_unit_code, '[^:]+:([0-9]{2}):*[^:]*') state_fips_code,
+               substring(prime.governmental_unit_code, '[^:]+:[^:]+:([0-9]{3})') county_fips_code,
```
Comment (ssoper-usgs) [inline src/main/resources/mybatis/station.xml]: Are we consolidating the huc values into one column and the fips codes into one column and then regex-ing out the values?

File: src/test/java/gov/usgs/cida/wqp/PostgresqlDataTypeFactoryWithJson.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+public class PostgresqlDataTypeFactoryWithJson extends PostgresqlDataTypeFactory {
+	private static final Logger LOG = LoggerFactory.getLogger(PostgresqlDataTypeFactoryWithJson.class);
+
+	@Override
+    public DataType createDataType(int sqlType, String sqlTypeName) throws DataTypeException {
+		LOG.debug("createDataType(sqlType={}, sqlTypeName={})",
+                String.valueOf(sqlType), sqlTypeName);
+
+        if (sqlType == Types.OTHER)
```
Comment (ssoper-usgs) [inline src/test/java/gov/usgs/cida/wqp/PostgresqlDataTypeFactoryWithJson.java]: This is confusing me - should there be curly braces on line 25 and lines 29-40?

File: src/main/java/gov/usgs/cida/wqp/dao/WqpArrayTypeHandler.java [hunk 3]
```diff
@@ -1,41 +1,35 @@
...
 	public void setNonNullParameter(PreparedStatement ps, int i, List<String> parameter, JdbcType jdbcType) throws SQLException {
-		Connection c = ps.getConnection();
-		if (c instanceof DelegatingConnection) {
-			Connection pooledConnection = c;
-			c = ((DelegatingConnection<?>)c).getInnermostDelegate();
-			if (c == null) {
-				throw new RuntimeException("Database resource pool configuration error: " + pooledConnection.getClass().getName());
-			}
-		}
+//		Connection c = ps.getConnection();
```
[the remark this answers is not in the document]
Comment (dsteinich) [reply src/main/java/gov/usgs/cida/wqp/dao/WqpArrayTypeHandler.java]: no, they should have been deleted.

File: src/main/java/gov/usgs/cida/wqp/swagger/SwaggerConfig.java [hunk 4]
```diff
@@ -147,10 +151,36 @@ private ApiKey apiKey() {
 	}
 
 	@Bean
-	public UiConfiguration uiConfig() {
+	public Object uiConfig() {
 		//This is needed in swagger 2 for the "try it" button on head requests - should not be needed with swagger 3
 		//It is needed in all WQP projects!!!
-		return new UiConfiguration(null, "none", "alpha", "schema", new String[] { "get", "post", "head" }, false, true, null);
+//		new UiConfiguration(null, "none", "alpha", "schema", new String[] { "get", "post", "head" }, false, true, null);
```
[the remark this answers is not in the document]
Comment (dsteinich) [reply src/main/java/gov/usgs/cida/wqp/swagger/SwaggerConfig.java]: no, they should have been deleted.

File: src/main/resources/mybatis/station.xml (hunk 1 above)
Comment (dsteinich) [reply src/main/resources/mybatis/station.xml]: They always were one field. Oracle had virtual columns that could be referenced and indexed. In PostgreSQL we don't have the virtual columns and need to migrate the regex from the database to the code.

File: src/test/java/gov/usgs/cida/wqp/PostgresqlDataTypeFactoryWithJson.java (hunk 2 above)
Comment (dsteinich) [reply src/test/java/gov/usgs/cida/wqp/PostgresqlDataTypeFactoryWithJson.java]: Whoops, delete happy... Interesting that it does appear to work...

File: src/main/resources/mybatis/station.xml (hunk 1 above)
Comment (ssoper-usgs) [reply src/main/resources/mybatis/station.xml]: Roger that, thanks for the clarifications!

File: src/test/java/gov/usgs/cida/wqp/PostgresqlDataTypeFactoryWithJson.java (hunk 2 above)
Comment (ssoper-usgs) [reply src/test/java/gov/usgs/cida/wqp/PostgresqlDataTypeFactoryWithJson.java]: Sounds good.

**Pull request merged**